app/src/main/java/com/jienihou/outfitxchange/productpage/ProductRVAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package com.jienihou.outfitxchange.productpage
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.jienihou.outfitxchange.R
+import com.jienihou.outfitxchange.data.ProductItem
+
+/**
+ * CategoryRVAdapter is the recyclerview adapter used for displaying categories
+ */
+class ProductRVAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    private var productItems: List<ProductItem> = ArrayList()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return ProductViewHolder(
+            LayoutInflater.from(parent.context).inflate(
+                R.layout.product_card_item,
+                parent,
+                false
+            )
+        )
+    }
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        if(holder is ProductViewHolder){
```

Comment:
Do you expect this to be any other type of view holder? If you expect the items to be homogenous (i.e all of the same type) you could probably benefit from getting rid of this type check and use a concrete type argument for `ProductRVAdapter`

`class ProductRVAdapter : RecyclerView.Adapter<ProductViewHolder>()`

Same comment applies to the other recycler view as well :)